utils/db.js

```diff
@@ -18,3 +18,15 @@ exports.checkArticleExists = (article_id) => {
     
     });
     }
+
+    exports.checkUserExists = (username) => {
+        return pool.query(
+            'SELECT * FROM articles WHERE author =$1',[username]).then((data)=>{
+                if(data.rows.length === 0){
+                    return Promise.reject({
+                        status: 404,
+                        msg: "Username does not exist"
+                    })
+                }
+        });
+        }
```

Comment:
You dont need this anymore since you are checking this by dealing with foreign key errors :)